File: backend/slotify/spring/src/main/java/com/slotify/backend/spring/service/dtos/PatchScheduleRequest.java
package com.slotify.backend.spring.service.dtos;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.openapitools.jackson.nullable.JsonNullable;

import java.time.LocalTime;

@Getter
@Setter
@NoArgsConstructor
public class PatchScheduleRequest {
    private JsonNullable<Integer> dayOfWeek = JsonNullable.undefined();
    private JsonNullable<LocalTime> startTime = JsonNullable.undefined();
    private JsonNullable<LocalTime> endTime = JsonNullable.undefined();
}

File: frontend/app/src/main/java/com/raj/slotify/dtos/service/PatchScheduleRequest.kt
package com.raj.slotify.dtos.service

import com.google.gson.annotations.SerializedName

data class PatchScheduleRequest(

    @SerializedName("dayOfWeek")
    val dayOfWeek: Int? = null,

    @SerializedName("startTime")
    val startTime: String? = null,

    @SerializedName("endTime")
    val endTime: String? = null

)
